app/views/Moderation/Reports/Window/index.tsx

```diff
@@ -0,0 +1,203 @@
+import React from 'react';
+import {
+    BarChart,
+    Bar,
+    XAxis,
+    YAxis,
+    Tooltip,
+    Legend,
+    ResponsiveContainer,
+    LineChart,
+    Line,
+    ScatterChart,
+    Scatter,
+    ZAxis,
+} from 'recharts';
+import { Container } from '@the-deep/deep-ui';
+import { ReportsQuery } from '#generated/types';
+
+import styles from './styles.css';
+
+type paymentPerOrderWindowType = NonNullable<NonNullable<ReportsQuery['moderatorQuery']>['reports']>['paymentPerOrderWindow'];
+type bookGradesPerOrderWindowType = NonNullable<NonNullable<ReportsQuery['moderatorQuery']>['reports']>['bookGradesPerOrderWindow'];
+type booksAndCostPerSchoolType = NonNullable<NonNullable<ReportsQuery['moderatorQuery']>['reports']>['booksAndCostPerSchool'];
+interface windowProps {
+    paymentPerOrderWindow: paymentPerOrderWindowType;
+    bookGradesPerOrderWindow: bookGradesPerOrderWindowType;
+    booksAndCostPerSchool: booksAndCostPerSchoolType;
+    orderWindows: {
+        window: string;
+        fill: string;
+    }[] | undefined
+}
+
+interface GradeOrder {
+    [key: string]: number | string;
+    grade: string;
+}
+
+function Window(props: windowProps) {
+    const { paymentPerOrderWindow,
+        bookGradesPerOrderWindow, orderWindows, booksAndCostPerSchool } = props;
+
+    const gradesOrder = bookGradesPerOrderWindow?.map((bg) => (
+        bg?.grades?.map((g) => ({ grade: g.grade, [bg.title]: g.numberOfBooks })) ?? []
+    ));
```

Comment:
we could make use of proper naming for variables.

Reply:
done